### PR TITLE
Feature/palo alto syslog sso

### PR DESCRIPTION
**Pull request by julsemaan**
# Description
Support syslog based SSO in Palo Alto firewall

# Impacts
PaloAlto SSO

# NEW Package(s) required
Net::Syslog

# Delete branch after merge
YES

# NEWS file entries
(REQUIRED, but may be optional...)
## Enhancements
* Added syslog transport method for PaloAlto SSO

### CODE REVIEW

**Comment by julsemaan** [comment]
code is ready for review, do not merge before doc is added

**Comment by julsemaan** [comment]
perl-Net-Syslog rpm was added to the devel RHEL6 + RHEL7 repos

Debian package seems available (https://packages.debian.org/stretch/libnet-syslogd-perl) but no clue how to add it. Will leave it to the pros

**Comment by julsemaan** [comment]
ready for review

**Comment by fdurand** [comment]
I will take a look for the Debian package.

**Comment by fdurand** [comment]
Package done for wheezy and jessie

File: lib/pf/constants/firewallsso.pm [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+
+use strict;
+use warnings;
+use base qw(Exporter);
+use Readonly;
+
+our @EXPORT_OK = qw($SYSLOG_TRANSPORT $HTTP_TRANSPORT);
+
+Readonly::Scalar our $SYSLOG_TRANSPORT => "syslog";
+Readonly::Scalar our $HTTP_TRANSPORT => "accounting";
```
Comment (fdurand) [inline lib/pf/constants/firewallsso.pm]: Why accounting ?

File: lib/pf/constants/firewallsso.pm [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+
+=head1 AUTHOR
+
+Inverse inc. <info@inverse.ca>
+
+=head1 COPYRIGHT
+
+Copyright (C) 2005-2016 Inverse inc.
```
Comment (fdurand) [inline lib/pf/constants/firewallsso.pm]: 2017

**Comment by fdurand** [comment]
Need to update RHEL/Debian packaging with the new module (Net-Syslog).

**Comment by julsemaan** [comment]
ready for another review

File: lib/pf/firewallsso/PaloAlto.pm [hunk 3]
```diff
@@ -125,6 +96,96 @@ sub logger {
...
+            </payload>
+       </uid-message>
+XML
+    my $webpage = "https://".$firewall_conf."/api/?type=user-id&action=set&key=".$ConfigFirewallSSO{$firewall_conf}->{'password'};
+    my $ua = LWP::UserAgent->new;
+    $ua->timeout(5);
+    my $response = $ua->post($webpage, Content => [ cmd => $message ]);
+    if ($response->is_success) {
+        $logger->info("Node $mac registered and allowed to pass the Firewall");
+        return 1;
```
Comment (fdurand) [inline lib/pf/firewallsso/PaloAlto.pm]: $TRUE

File: lib/pf/firewallsso/PaloAlto.pm [hunk 4]
```diff
@@ -125,6 +96,96 @@ sub logger {
...
+    } else {
+        $logger->error("XML send error :".$response->status_line);
+        return 0;
```
Comment (fdurand) [inline lib/pf/firewallsso/PaloAlto.pm]: $FALSE

File: lib/pf/firewallsso/PaloAlto.pm [hunk 5]
```diff
@@ -125,6 +96,96 @@ sub logger {
...
+            </payload>
+       </uid-message>
+XML
+    my $webpage = "https://".$firewall_conf."/api/?type=user-id&action=set&key=".$ConfigFirewallSSO{$firewall_conf}->{'password'};
+    my $ua = LWP::UserAgent->new;
+    $ua->timeout(5);
+    my $response = $ua->post($webpage, Content => [ cmd => $message ]);
+    if ($response->is_success) {
+        $logger->debug("Node $mac removed from the firewall");
+        return 1;
```
Comment (fdurand) [inline lib/pf/firewallsso/PaloAlto.pm]: $TRUE

File: lib/pf/firewallsso/PaloAlto.pm [hunk 6]
```diff
@@ -125,6 +96,96 @@ sub logger {
...
+    } else {
+        $logger->error("XML send error :".$response->status_line);
+        return 0;
```
Comment (fdurand) [inline lib/pf/firewallsso/PaloAlto.pm]: $FALSE

File: docs/PacketFence_PaloAlto_Quick_Install_Guide.asciidoc [hunk 7]
```diff
@@ -90,3 +92,51 @@ IP              Vsys   From    User                             IdleTimeout(s) M
 --------------- ------ ------- -------------------------------- -------------- -------------
 192.168.100.10  vsys1  XMLAPI  domain\user1                     Never          Never     
 ----
+
+Installation using syslog
+-------------------------
+
+NOTE: This installation mode is not suggested unless you use the SSO for informationnal purposes (no enforcement). PacketFence will use easily spoofable UDP packets to communicate with the Palo Alto firewall. If you require encryption and origin validation of the SSO messages, please use the XML API.
```
Comment (louismunro) [inline docs/PacketFence_PaloAlto_Quick_Install_Guide.asciidoc]: s/informationnal/informational/

**Comment by louismunro** [comment]
Please make the minor changes asked by @fdurand and we can merge.

**Comment by julsemaan** [comment]
Adjustments done

**Pull request merged**